server/src/main/java/io/druid/server/initialization/jetty/JettyServerModule.java

```diff
@@ -161,6 +162,10 @@ private static Server makeJettyServer(@Self DruidNode node, ServerConfig config)
 
     final Server server = new Server(threadPool);
 
+    // Without this bean set, the default ScheduledExecutorScheduler runs as non-daemon, causing lifecycle hooks to fail
+    // to fire on main exit. Related bug: https://github.com/druid-io/druid/pull/1627
+    server.addBean(new ScheduledExecutorScheduler("JettyScheduler", true), true);
```

Comment:
@himanshug  If the stop is called then yes. The problem is that if you let io.druid.cli.Main naturally exit, then the lifecycle.stop which calls server.stop is never called because this is a daemon thread. With this patch, a clean exit out of Main can be accomplished because the jetty scheduler is not holding up the shutdown hook